### PR TITLE
DDCE-5031 Changes for Play 3.0 migrations

### PR DESCRIPTION
**Pull request by mywyau**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
-    "com.amazonaws"                 % "aws-java-sdk-s3"           % "1.12.521",
-    "uk.gov.hmrc"                  %% "bootstrap-backend-play-28" % bootstrapPlayVersion,
-    "uk.gov.hmrc.mongo"            %% "hmrc-mongo-play-28"        % mongoHmrcVersion,
-    "uk.gov.hmrc"                  %% "play-json-union-formatter" % "1.18.0-play-28",
-    "com.fasterxml.jackson.module" %% "jackson-module-scala"      % "2.15.2",
+    "com.amazonaws"                 % "aws-java-sdk-s3"           % "1.12.683",
+    "uk.gov.hmrc"                  %% "play-json-union-formatter" % "1.21.0",
     "org.apache.httpcomponents"     % "httpclient"                % apacheHttpVersion,
-    "org.apache.httpcomponents"     % "httpmime"                  % apacheHttpVersion
+    "org.apache.httpcomponents"     % "httpmime"                  % apacheHttpVersion,
```
Comment (EdCope) [inline project/AppDependencies.scala:14]: Like Scalaj we agreed to remove other http clients. This one seems to be only used in one test

**Review by EdCope — CHANGES_REQUESTED** [review]

**Review by EdCope — CHANGES_REQUESTED** [review]

File: project/CodeCoverageSettings.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import sbt.Setting
+import scoverage.ScoverageKeys.*
+
+object CodeCoverageSettings {
+
+  private val excludedPackages: Seq[String] = Seq(
+    "<empty>",
```
[the remark this answers is not in the document]
Comment (mywyau) [reply project/CodeCoverageSettings.scala:7]: Screenshot 2024-03-28 at 10 06 01

[the remark this answers is not in the document]
Comment (Taiwo-AB) [reply project/CodeCoverageSettings.scala:7]: If you want to take out empty then you should use `coverageExcludedFiles := ".*Routes.*"` and delete the val.

[the remark this answers is not in the document]
Comment (EdCope) [reply project/CodeCoverageSettings.scala:7]: Sorry, i didn't realise it was doing this to `FileStoreController`. What Taiwo said is an option or can leave then

File: it/test/resources/upscan/initiate_response.json
[the remark this answers is not in the document]
Comment (mywyau) [reply it/test/resources/upscan/initiate_response.json:1]: think this has been cleaned up now, deleted some files and removed the ResourceFiles trait

File: project/CodeCoverageSettings.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (mywyau) [reply project/CodeCoverageSettings.scala:7]: no worries kept checking it and leaving it in and forgot to leave a message my bad.

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -1,32 +1,26 @@
...
-    "com.amazonaws"                 % "aws-java-sdk-s3"           % "1.12.521",
-    "uk.gov.hmrc"                  %% "bootstrap-backend-play-28" % bootstrapPlayVersion,
-    "uk.gov.hmrc.mongo"            %% "hmrc-mongo-play-28"        % mongoHmrcVersion,
-    "uk.gov.hmrc"                  %% "play-json-union-formatter" % "1.18.0-play-28",
-    "com.fasterxml.jackson.module" %% "jackson-module-scala"      % "2.15.2",
-    "org.apache.httpcomponents"     % "httpclient"                % apacheHttpVersion,
-    "org.apache.httpcomponents"     % "httpmime"                  % apacheHttpVersion
+    "com.amazonaws"                 % "aws-java-sdk-s3"           % "1.12.683",
+    "uk.gov.hmrc"                  %% "play-json-union-formatter" % "1.21.0",
+    "org.apache.httpcomponents"     % "httpmime"                  % apacheHttpVersion,
```
[the remark this answers is not in the document]
Comment (mywyau) [reply project/AppDependencies.scala:13]: I think we should leave it as extra work and make a new ticket. No idea on what the rewrite would look like. And should just get the migration over first.

**Review by EdCope — APPROVED** [review]

**Pull request merged**